coalib/output/ConsoleInteraction.py

```diff
@@ -93,6 +94,8 @@ def highlight_text(no_color, text, style, lexer=TextLexer()):
                ShowAppliedPatchesAction(),
                GeneratePatchesAction())
 DIFF_EXCERPT_MAX_SIZE = 4
+APPLY_SINGLE = 1
+APPLY_SINGLE_ACTION = 'NONE'
```

Comment:
hmmmm... or you can modify `Processing.py` so that the responsible queue of methods which actually call `choose_action` gets called only once if `apply_single` is True ov🤔er there

Reply:
seems better than maintaining these global variables

Reply:
Well, choose_action is actually called the number of times the patches needed to be applied, if i make changes in the processing.py to call choose_action only once then only patch will get applied.

Reply:
but you have access to `apply_single` in `Processing.py` right?

Reply:
Yes we do. But it doesn't control how the patches are applied i think. I don't know i might be wrong. But as much as i experimented, irrespective of whether it is a apply_single action or not, choose_action() method is being called everytime a patch has to be applied. choose_action method is also responsible for passing the selected action back to processing.py.

Reply:
When apply_single is selected the action is stored in a variable and is checked with the available actions everytime in the choose_Action and passes it back to process. If we want to change that then i think we need to write a whole new function for it and make lots of changes :sweat_smile: